hanadb_exporter/exporters/prometheus_exporter.py

```diff
@@ -46,33 +46,33 @@ def _execute(self, metric):
         try:
             value = self._hdb_connector.query(metric['query'])
 
-            if metric['type'] == core.GaugeMetricFamily:
-                metric_obj = self._manage_gauge(metric, value)
+            if metric['type'] == "core.GaugeMetricFamily":
+                metric_obj = self._manage_gauge(metric_name, metric, value)
             else:
                 raise NotImplementedError('{} type not implemented'.format(metric['type']))
 
             return metric_obj
         except KeyError as err:
             raise MalformedMetric(err)
 
-    def _manage_gauge(self, metric, value):
+    def _manage_gauge(self, metric_name, metric, value):
         """
         Manage Gauge type metric
         """
         # Label not set
-        metric_obj = core.GaugeMetricFamily(*metric['info'])
-        if not metric['info'][3]:
-            metric_obj.add_metric([], str(value[0][0]))
-        else:
-            for label_item in value:
-                self._logger.info('%s: %s' % (label_item[0], label_item[1]))
-                metric_obj.add_metric([label_item[0]], float(label_item[1]))
+        metric_obj = core.GaugeMetricFamily(metric_name, metric['description'], None, metric['labels'], metric['unit'])
+
+        metric_obj.add_metric(metric['labels'], str(value[0][-1]))
+        for label_item in value:
+            self._logger.info('%s' % (label_item[0]))
+
         return metric_obj
 
     def collect(self):
         """
         Collect data from database
         """
-        for metric in METRICS:
-            metric_obj = self._execute(metric)
+        metrics = PrometheusMetrics()
+        for (k, v) in metrics.data.items():
```

Comment:
Parenthesis are not needed here.
Use a better defined instead of `k` and `v` like `metric_name` and `metric_data`